### PR TITLE
feat: new homepage

### PR DESCRIPTION
**Pull request by TheChronicMonster**
Introducing the new docs homepage

git moved existing homepage to `docs/dev/bak/exp/index.md`

Created new `docs/src/index.md` with new copy

Added `DISCUSSIONS` to `src/config.json` to call with @{DISCUSSIONS}

Added `why` directory with images making the case for developer adoption
`why` `index.md` displaying images and `config.json` to order place in TOC

### CODE REVIEW

File: docs/src/assets/icons/Colorway=Default, State=Default.svg:Zone.Identifier [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[ZoneTransfer]
+ZoneId=3
+ReferrerUrl=C:\Users\hamir\Downloads\Doc Icons.zip
```
Comment (jeapostrophe) [inline docs/src/assets/icons/Colorway=Default, State=Default.svg:Zone.Identifier:3]: What is the deal with these files that refer to Hamir's machine?

**Review by jeapostrophe — CHANGES_REQUESTED** [review]
Why do we need all the different flags?

**Comment by TheChronicMonster** [comment]
> l

Flags are for the future; removed

File: docs/src/assets/icons/Colorway=Default, State=Default.svg:Zone.Identifier (hunk 1 above)
Comment (TheChronicMonster) [reply docs/src/assets/icons/Colorway=Default, State=Default.svg:Zone.Identifier]: I wondered about that, as well. They're gone now.

**Comment by TheChronicMonster** [comment]
I resolved conflicting files on my remote, but am receiving the messge
```
remote: error: GH006: Protected branch update failed for refs/heads/master.
remote: error: At least 1 approving review is required by reviewers with write access.
To github.com:reach-sh/reach-lang.git
```

**Comment by jeapostrophe** [comment]
You need to rebase

**Comment by TheChronicMonster** [comment]
> You need to rebase

Believe I have earned my rebase badge.
PR ready pending your approval

File: docs/src/assets/scripts.js [hunk 2]
```diff
@@ -29,45 +29,66 @@ const maxColWidth = '280px';
 let winWidth = getWinWidthStr();
 
 const establishDisplay = () => {
-  const { bookPath, hasOtp } = currentPage;
```
Comment (jeapostrophe) [inline docs/src/assets/scripts.js:32]: You merged this file incorrectly and blew away some important changes

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: docs/src/assets/scripts.js (hunk 2 above)
Comment (TheChronicMonster) [reply docs/src/assets/scripts.js:32]: When I kept those changes the bookPath in the left path no longer hid. 
I saved the revisions that weren't merged.

Comment (jeapostrophe) [reply docs/src/assets/scripts.js:32]: You'll have to have Hamir fix it then, because the changes are important

Comment (jeapostrophe) [reply docs/src/assets/scripts.js:32]: This file is still wrong. It doesn't have the changes he's made lately... it has dropped the abstraction of displayFor and it has dropped the Github Discussions search entries

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: docs/xrefs.json [hunk 3]
```diff
@@ -707,6 +707,10 @@
     "title": "RX0000",
     "path": "/rsh/errors/#RX0000"
   },
+  "build": {
+    "title": "Reach is for Builders",
+    "path": "/build/#build"
```
Comment (hamirmahal) [inline docs/xrefs.json:712]: This pull request needs the `build/` directory.

**Review by hamirmahal — CHANGES_REQUESTED** [review]

File: docs/xrefs.json (hunk 3 above)
Comment (TheChronicMonster) [reply docs/xrefs.json:712]: `build/` has been restored.

**Review by jeapostrophe — APPROVED** [review]

**Comment by jeapostrophe** [comment]
I'm letting you merge, so you can (a) control when it goes live and (b) ensure that you're happy with the look

**Pull request merged**